packages/core/src/components/button/button.stories.ts

```diff
@@ -0,0 +1,59 @@
+import { htmlMatrix, Meta, omit, Story } from '../../../../../docs';
+import { Icon } from '../icon/icon.story';
+import { Button, ButtonProps } from './button.story';
+
+export default {
+  title: 'Core/Button',
+  component: Button,
+  argTypes: {
+    children: {},
+    disabled: {
+      description: 'Only available in `<button>` element.',
+    },
+    href: {
+      description: 'Use with `<a>` element instead of `<button>`.',
+    },
+  },
+  args: {
+    children: 'Button',
+    disabled: false,
+    kind: 'action',
+    variant: 'primary',
+  },
+} as Meta<ButtonProps>;
+
+const disabled = [true, false] as const;
+const kind = ['action', 'destructive'] as const;
+const variant = ['primary', 'secondary', 'tertiary'] as const;
+
+export const Playground: Story<ButtonProps> = (props) => Button(props);
+
+export const Kind = htmlMatrix(Button, { kind });
+Kind.argTypes = omit<ButtonProps>('kind', 'children');
+
+export const Variant = htmlMatrix(Button, { variant });
+Variant.argTypes = omit<ButtonProps>('variant', 'children');
+
+export const Disabled = htmlMatrix(Button, { disabled });
+Disabled.argTypes = omit<ButtonProps>('disabled', 'children');
+
+export const AsAnchor: Story<ButtonProps> = (props) =>
+  Button({ ...props, href: 'javascript:void 0' });
+
+const icon = Icon({ name: 'alarm' });
```

Comment:
To make sure first icon is selected from available, also would always match React story:

```suggestion
const [firstIconName] = iconNames;
const icon = Icon({ name: firstIconName });
```